tests/model/schema.js

```diff
@@ -641,6 +641,48 @@ describe( 'Schema', () => {
 		} );
 	} );
 
+	describe( 'checkMerge()', () => {
+		beforeEach( () => {
+			schema.register( '$root' );
+			schema.register( '$block', {
+				allowIn: '$root',
+				isBlock: true
+			} );
+			schema.register( '$text', {
+				allowIn: '$block'
+			} );
+			schema.register( 'paragraph', {
+				allowWhere: '$block'
+			} );
+			schema.register( 'blockQuote', {
+				allowWhere: '$block',
+				allowContentOf: '$root'
+			} );
+			schema.register( 'listItem', {
+				inheritAllFrom: '$block'
+			} );
+		} );
+
+		it( 'returns false if a block cannot be merged with other block', () => {
+			const paragraph = new Element( 'paragraph', null, [
+				new Text( 'xyz' )
+			] );
+			const blockQuote = new Element( 'blockQuote', null, [ paragraph ] );
+			const listItem = new Element( 'listItem' );
+
+			expect( schema.checkMerge( listItem, blockQuote ) ).to.equal( false );
+		} );
+
+		it( 'returns true if a block can be merged with other block', () => {
+			const listItem = new Element( 'listItem' );
+			const listItemToMerge = new Element( 'listItem', null, [
+				new Text( 'xyz' )
+			] );
+
+			expect( schema.checkMerge( listItem, listItemToMerge ) ).to.equal( true );
+		} );
```

Comment:
Please add a test where `elementToMerge` has more children and only one of them (not the first one) will not be allowed in the `baseElement`.

Also, another test where `baseElement` already contains disallowed elements, but `elementToMerge` contains only allowed ones. This test will verify stability of the implementation. Such situation is by definition invalid, but current (sufficient) implementation of `checkMerge()` wouldn't reject it, so let's be sure that future implementations won't do that too.

Reply:
Missing tests have been added in ddf9e54.